### PR TITLE
[PL-50674] DEL 82900 Release Notes

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
[PL-50674](https://harness.atlassian.net/browse/PL-50674)

https://harness.atlassian.net/wiki/spaces/PD/pages/21768012211/NG+Release+SAAS+April+29th+2024+for+1.36.x-+Build#NG-Delegate-dependent-changes--

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

[PL-50674]: https://harness.atlassian.net/browse/PL-50674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: release-notes/delegate.md [hunk 1]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 <Deleos />
 
+## May 2024
+
+### Harness version 1.36.x, Harness Delegate version 24.04.82900 <!--  May xx, 2024 -->
```
Comment (brian-f-harness) [inline release-notes/delegate.md:42]:
```suggestion
### Version 24.04.82900 <!--  May xx, 2024 -->
```

File: release-notes/delegate.md [hunk 2]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
```
Comment (brian-f-harness) [inline release-notes/delegate.md:43]: We no longer want to have the Harness version in the heading.

https://harness.slack.com/archives/C06V7T3UTBN/p1713925064939699

File: release-notes/delegate.md [hunk 3]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+#### Fixed issues
+
+Delegates were incorrectly displayed as connected despite being configured with non-agent endpoints. Resolved this by ensuring the isNg flag is correctly propagated when delegates send heartbeats to the manager, ensuring accurate connection status. (PL-48891, ZD-60974)
+
+Intermittent Socket Timeout Exceptions were occurring in running pipelines due to secret decryption failures, triggering unnecessary re-broadcasts on the delegate side. We have addressed the issue of intermittent secret decryption failures within pipelines, ensuring stable and uninterrupted pipeline execution. (PL-47940, ZD-58006)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:48]:
```suggestion
Intermittent Socket Timeout Exceptions occurred in running pipelines due to secret decryption failures, triggering unnecessary re-broadcasts on the delegate side. Resolved the issue of intermittent secret decryption failures within pipelines, ensuring stable and uninterrupted pipeline execution. (PL-47940, ZD-58006)
```

File: release-notes/delegate.md [hunk 4]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+- Local login was failing for users assigned admin permissions via a User Group. The method to verify if a user is an account admin only considered direct user assignments and did not account for User Group roles. We have revised the validation process to include both User and User Group assignments when checking for admin status. Moving forward, to be recognized as an admin, users must have specific Role assignments outlined below; merely assigning the _account_admin role will no longer suffice for admin recognition. (PL-47632)
+   Role : _account_admin
+   Resource-group : _all_resources_including_child_scopes, _all_account_level_resources.
```
Comment (brian-f-harness) [inline release-notes/delegate.md:52]: If this is code, add a code block

File: release-notes/delegate.md [hunk 5]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+
+#### Fixed issues
+
+Delegates were incorrectly displayed as connected despite being configured with non-agent endpoints. Resolved this by ensuring the isNg flag is correctly propagated when delegates send heartbeats to the manager, ensuring accurate connection status. (PL-48891, ZD-60974)
+
+Intermittent Socket Timeout Exceptions were occurring in running pipelines due to secret decryption failures, triggering unnecessary re-broadcasts on the delegate side. We have addressed the issue of intermittent secret decryption failures within pipelines, ensuring stable and uninterrupted pipeline execution. (PL-47940, ZD-58006)
+
+- Local login was failing for users assigned admin permissions via a User Group. The method to verify if a user is an account admin only considered direct user assignments and did not account for User Group roles. We have revised the validation process to include both User and User Group assignments when checking for admin status. Moving forward, to be recognized as an admin, users must have specific Role assignments outlined below; merely assigning the _account_admin role will no longer suffice for admin recognition. (PL-47632)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:50]:
```suggestion
- Local login was failing for users assigned admin permissions via a user group. The method to verify if a user is an account admin only considered direct user assignments and did not account for user group roles. Revised the validation process to include both user and user group assignments when checking for admin status. Now, to be recognized as an admin, users must have the specific role assignments outlined below; assigning the `_account_admin` role alone is no longer sufficient for admin rights. (PL-47632)
```

File: release-notes/delegate.md [hunk 6]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
+   Role : _account_admin
```
Comment (brian-f-harness) [inline release-notes/delegate.md:51]:
```suggestion
   - Role : `_account_admin`
```

File: release-notes/delegate.md (hunk 4 above)
Comment (brian-f-harness) [inline release-notes/delegate.md:52]:
```suggestion
   - Resource-group : `_all_resources_including_child_scopes`, `_all_account_level_resources`.
```

File: release-notes/delegate.md [hunk 7]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
...
 
+## May 2024
+
+### Version 24.04.82900 <!--  May xx, 2024 -->
+
+#### Fixed issues
+
+Delegates were incorrectly displayed as connected despite being configured with non-agent endpoints. Resolved this by ensuring the `isNg` flag is correctly propagated when delegates send heartbeats to the manager, ensuring accurate connection status. (PL-48891, ZD-60974)
+
+Intermittent Socket Timeout Exceptions occurred in running pipelines due to secret decryption failures, triggering unnecessary re-broadcasts on the delegate side. Resolved the issue of intermittent secret decryption failures within pipelines, ensuring stable and uninterrupted pipeline execution. (PL-47940, ZD-58006)
```
Comment (brian-f-harness) [inline release-notes/delegate.md:48]:
```suggestion
Intermittent socket timeout exceptions occurred in running pipelines due to secret decryption failures, triggering unnecessary re-broadcasts on the delegate side. Resolved the issue of intermittent secret decryption failures within pipelines, ensuring stable and uninterrupted pipeline execution. (PL-47940, ZD-58006)
```

File: release-notes/delegate.md [hunk 8]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 <Deleos />
 
+## May 2024
+
+### Version 24.04.82900 <!--  May xx, 2024 -->
+
```
Comment (brian-f-harness) [inline release-notes/delegate.md:43]: You can find the delegate version that we will deploy (in this case 24.04.82901) when the PL deployment is announced in the #prod-announce Slack channel.
https://harness.slack.com/archives/C04R4UDE1FT/p1714972253677999

File: release-notes/delegate.md [hunk 9]
```diff
@@ -37,6 +37,20 @@ import Deleos from '/docs/platform/shared/delegate-legacy-eos.md'
 
 <Deleos />
 
+## May 2024
+
+### Version 24.04.82901 <!--  May 8, 2024 -->
```
Comment (brian-f-harness) [inline release-notes/delegate.md:42]: Also need to update the date in the frontmatter for the RSS feed
date: 2024-05-08T10:00

Comment (saifeemustafaqharness) [reply release-notes/delegate.md:42]: Done. Thanks

**Comment by brian-f-harness** [comment]
@saifeemustafaqharness There's a merge conflict that needs to be resolved.

**Review by brian-f-harness — APPROVED** [review]

**Review by pratmit — APPROVED** [review]
LGTM

**Review by douglas-j-bothwell — APPROVED** [review]
LGTM no changes

**Pull request merged**